### ISSUE TITLE
Develop branch, fsi_computation.py error using --parallel (with parallel building of the wrapper)

### ISSUE THREAD

**Issue by RoccoBombardieri** [su2code/SU2#531]
Hello,

I'm experiencing an unclear error trying to launch the fsi_computation.py of the **SU2 develop branch**. From the command line, launching:

`mpirun -np XX fsi_computation.py -f FSICoupler_config.cfg --parallel`

with XX number of process (I tried both with 1 and more), I get the error:
> ERROR : You are trying to initialize MPI with a serial build of the wrapper. Please, remove the --parallel option that is incompatible with a serial build.

I'm sure my wrapper has been built in parallel, as I used the configure options suggested in 
[https://su2code.github.io/docs/Python-Wrapper-Build/](url)
Plus, I'm doing exactly the same as done using the old branch **feature_pyWrapper** which, according to my old issue:
[https://github.com/su2code/SU2/issues/449#issuecomment-333879558](url)
should have been merged to the develop branch.
Is this a code problem or I'm doing something wrong?  (Am I referring to the correct branch?)

Thanks in advance for the suggestions

EDIT: more in details the exception raised is 

> ('A TypeError occured in pysu2.CSingleZoneDriver : ', TypeError('__init__() takes exactly 6 arguments (5 given)',))

**Comment by tobadavid** [comment]
Hi,

This error simply occurs because the prototype of the CDriver (and child classes) has changed. There is an additional `bool val_periodic` argument that have to be passed. The current fsi_computation.py is not up-to-date. Here are solutions to fix that:

- Change the fsi_computation.py and pass an extra `False` argument between the number of dimensions and the communicator.

or

- Do you need the very last develop branch ? Because it seems like in the master branch the constructor of the CDriver is still the compatible one. So you could use this branch without modifying anything.

I hope this will help.

Cheers,

David

**Comment by RoccoBombardieri** [comment]
Hi David,
Yes, I was able to figure out that and I went ahead in trying to make fsi_computation.py work with this branch. Another error (which may be a bug actually ) occurs when the wrapped function **SetInitialMesh()** is run. This directly involves the core of SU2 as I found out the error is given during the run of the function **CILUPreconditioner** (inside **matrix_structure.cpp**) which, as much I understood, builds the preconditioning matrix for the solution of the linear system at the base of the mesh deforming procedure. 

SU2 uses by default **ILU** precoditioner whose relative function (see above) has remarkable differences from the Raven release (I didn't go too much into details).  Same error I get if using **Jacobi** preconditioner. 
At contrary, everything seems fine if using **LU_SGS** preconditioner and the fsi_computation.py finally works. Maybe it's the case to point it out to the developers inveloved in this part of the code.

About your last point, I'm developing at the moment some FSI features and I very much relied on fsi_computation.py in defining the interface with the SU2 core. I'm trying to keep it updated with the current version of SU2 for reliability reasons; I hope there won't be drastic changes in the python wrapping of the code! :)

Thanks for the reply!

Rocco
